cmd/auth-provider-gcp/app/getcredentials.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/spf13/cobra"
+	utilnet "k8s.io/apimachinery/pkg/util/net"
+	"k8s.io/cloud-provider-gcp/cmd/auth-provider-gcp/credentialconfig"
+	"k8s.io/cloud-provider-gcp/cmd/auth-provider-gcp/plugin"
+	klog "k8s.io/klog/v2"
+	"net/http"
+)
+
+var (
+	authFlow string
+)
+
+const (
+	gcrAuthFlow             = "gcr"
+	dockerConfigAuthFlow    = "dockercfg"
+	dockerConfigURLAuthFlow = "dockercfg-url"
+)
+
+// NewGetCredentialsCommand returns a cobra command that retrieves auth credentials after validating flags.
+func NewGetCredentialsCommand() (*cobra.Command, error) {
+	cmd := &cobra.Command{
+		Use:   "get-credentials",
```

Comment:
I think I'm a little confused on how `Use` should be applied with sub-commands. Since this is a sub-command and will end up getting used on the command line with something like `auth-provider-gcp get-credentials`, should this string also read `auth-provider-gcp`? I think the root command already had the name of the binary mentioned, so should this describe exclusively flags and not mention the sub-command by name at all?

Reply:
I wrote the above before I realized this was a subcommand, so most of it applies to main.go.

Regardless, I don't think we should change anything in this PR. I'm just not happy with how cobra works.